### ISSUE TITLE
Unstable Metric API Causes Problems When Trying To Use This Module

### ISSUE THREAD

**Issue by hcelaloner** [couchbase/gocb-opentelemetry#5]
Hi, 

During OpenTelemetry adaptation, for Golang we saw this module and decided to give it a try. However, we are getting the following error when installing this module 

```
➜ go get github.com/couchbase/gocb-opentelemetry
go: downloading github.com/couchbase/gocb-opentelemetry v0.1.0
go: downloading go.opentelemetry.io/otel/metric v0.18.0
github.com/couchbase/gocb-opentelemetry imports
        go.opentelemetry.io/otel/metric imports
        go.opentelemetry.io/otel/unit: cannot find module providing package go.opentelemetry.io/otel/unit
```

I guess it is giving this error since in the latest releases metrics API changed. If that is the reason, can we fix this issue by using the latest metric API in this module?  Or not sure but it seems that JVM side has two separate modules [metrics-opentelemetry](https://github.com/couchbase/couchbase-jvm-clients/blob/master/metrics-opentelemetry/pom.xml) and [tracing-opentelemetry](https://github.com/couchbase/couchbase-jvm-clients/blob/master/tracing-opentelemetry/pom.xml). Should Golang side follow the same approach since metrics API is still unstable?

**Comment by jejikenwogu** [comment]
The latest commit works:
go get github.com/couchbase/gocb-opentelemetry@80b309d

They haven't updated the tag for some reason.

**Comment by chvck** [comment]
Apologies for the slow reply. I've just cut a 0.1.1 tag which should contains the fix(es) in 80b309d. 

> Should Golang side follow the same approach since metrics API is still unstable?

I've been considering doing this too, by creating 2 different modules within this repo so that we can release a stable tracing interface whilst leaving the metrics one unstable. This is has opentelemetry themselves do it too.

I think that this should now be fixed so I'll close this issue and I'll open a new one for splitting the module into two. Please reopen this one if you still see the same issue.
